Use Celo Sepolia for e2e test instead of alfajores

karlb opened the pull request:
The changed error message is more precise and more stable across geth versions.

Right now, the test "zero tip fee currency tx rejected" still fails for Sepolia, but the same is true for Alfajores. It will work once the node is updated. It is only run in CI when e2e test files are changed, so merging it in this state won't be too annoying.

Closes https://github.com/celo-org/celo-blockchain-planning/issues/1158

karlb committed:
Use Celo Sepolia for e2e test instead of alfajores

The changed error message is more precise and more stable across geth
versions.

Right now, the test "zero tip fee currency tx rejected" still fails for
Sepolia, but the same is true for Alfajores. It will work once the node
is updated.

Closes https://github.com/celo-org/celo-blockchain-planning/issues/1158

palango commented:
>  It is only run in CI when e2e test files are changed

Is this expected? Shouldn't we run this on some fixed schedule?

palango commented on e2e_test/js-tests/viem_setup.mjs:
```diff
@@ -35,6 +35,18 @@ const celoBaklava = defineChain({
 	},
 });
 
+const celoSepolia = defineChain({
```
Looks like we could inport `celoSepolia` directly: https://github.com/wevm/viem/blob/main/src/chains/index.ts#L97

karlb replied:
Done. To get it working I had to update viem and change an assert in our tests due to https://github.com/wevm/viem/pull/3117 being included in the viem update.

palango reviewed (approved):
Looks good, left one comment, but feel free to ignore.

karlb committed:
e2e: Update viem

This brings a celoSepolia definition and staying up to date is generally
a good idea.
Due to https://github.com/wevm/viem/pull/3117 being merged in view, we
can (and have to) remove a changed assert in the "test gas price
difference for fee currency" test.

palango commented:
@SocketSecurity ignore-all
